pkg/api/aim/dashboards.go

```diff
@@ -5,16 +5,27 @@ import (
 
 	"github.com/gofiber/fiber/v2"
 	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
 	"gorm.io/gorm"
 
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api"
+	"github.com/G-Research/fasttrackml/pkg/common/middleware/namespace"
 	"github.com/G-Research/fasttrackml/pkg/database"
 )
 
 func GetDashboards(c *fiber.Ctx) error {
+	ns, err := namespace.GetNamespaceFromContext(c.Context())
+	if err != nil {
+		return api.NewInternalError("error getting namespace from context")
+	}
+	log.Debugf("getDashboards namespace: %s", ns.Code)
+
 	var dashboards []database.Dashboard
 	if err := database.DB.
+		Preload("App").
+		Joins("LEFT JOIN apps ON apps.id = dashboards.app_id").
 		Where("NOT dashboards.is_archived").
-		Joins("App", database.DB.Select("ID", "Type", "IsArchived")).
+		Where("apps.namespace_id = ?", ns.ID).
 		Order("dashboards.updated_at").
 		Find(&dashboards).
```

Comment:
The whole SQL query building should be like this instead:

```go
		InnerJoins("App",
			database.DB.
				Select("ID", "Type").
				Where("namespace_id = ? AND NOT App.is_archived", ns.ID)).
		Where("NOT dashboards.is_archived").
		Order("dashboards.updated_at").
		Find(&dashboards).
```

This will ensure:
* we do a _single_ query to get both the dashboards and their associated apps
* we only get the information we need from the `apps` table
* we filter by namespace in the JOIN statement itself
* (new, we were not doing this before) we filter out archived apps

Reply:
probably the final result has to be:
```
InnerJoins(
			"App",
			database.DB.Select("ID", "Type").Where(`"App".namespace_id = ? AND NOT "App".is_archived`, ns.ID),
		)
```
it is why I tried to do not use this syntax.